ngraph/frontend/onnx/frontend/src/core/graph.cpp

```diff
@@ -105,6 +112,12 @@ Graph::Graph(std::shared_ptr<ONNX_NAMESPACE::ModelProto> model_proto, std::uniqu
         }
     }
 
+    if (telemetry) {
```

Comment:
Sending of the telemetry data is not really a part of the graph construction process. What do you think about extracting it out of the graph class and doing that after the graph has been constructed? You could still collect the statistics and expose them from the graph using a dedicated method. This way we'll be closer to the single responsibility principle for this constructor.

Reply:
@tomdol Yes, I thought about it too, but this approach has small 2 drawbacks: 
1. We should call a new send_telemetry() method immediately after Graph construction to avoid Exceptions/asserts that  can be thrown during convert/decode/transformation etc phases. Otherwise, we will not receive any information about this attempt to load the network via frontend.

And code will look like:
```
graph = Graph(...);
// telemetry should be called immediately after Graph construction.
graph->send_telemetry(m_telemetry);
graph->convert(..)
```
2. We'll have one more iteration over all graph nodes to just send Telemetry events. I hope it doesn't affect performance in general, but it looks more wasteful.

These points are not so significant and If this seems acceptable to you, then I will correct the implementation.

Reply:
Ok, I see your point. Let's leave it as is and consider splitting those 2 functionalities in the future.

Reply:
Ok!